src/settings.ts

```diff
@@ -1,59 +1,165 @@
+    }
+
+    getInt64(key: string): number {
+        return this._settings.get_int64(key);
+    }
+
+    getString(key: string): string {
+        return this._settings.get_string(key);
+    }
+
+    getStrv(key: string): string[] {
+        return this._settings.get_strv(key);
+    }
+
+    getSchema() {
+        return this._settings.settings_schema;
+    }
+
+    isWritable(key: string) {
+        return this._settings.is_writable(key);
+    }
+
+    listKeys() {
+        return this._settings.list_keys();
+    }
+
+    // eslint-disable-next-line no-unused-vars
+    observe(key: string, callback: (...args: any[]) => any) {
+        return this._settings.connect(`changed::${key}`, callback);
+    }
+
+    reset(keyName: string) {
+        this._settings.reset(keyName);
+    }
+
+    setBoolean(key: string, value: boolean) {
+        if (this._settings.set_boolean(key, value))
+            this._save();
+        else
+            throw new Error(`Could not set ${key} (type: boolean) with the value ${value}`);
+    }
+
+    setEnum(key: string, value: number) {
+        if (this._settings.set_enum(key, value))
+            this._save();
+        else
+            throw new Error(`Could not set ${key} (type: number) with the value ${value}`);
+    }
+
+    setInt64(key: string, value: number) {
+        if (this._settings.set_int64(key, value))
+            this._save();
+        else
+            throw new Error(`Could not set ${key} (type: number64) with the value ${value}`);
+    }
+
+    setString(key: string, value: string) {
+        if (this._settings.set_string(key, value))
+            this._save();
+        else
+            throw new Error(`Could not set ${key} (type: string) with the value ${value}`);
+    }
+
+    setStrv(key: string, value: string[]) {
+        if (this._settings.set_strv(key, value))
+            this._save();
+        else
+            throw new Error(`Could not set ${key} (type: string[]) with the value ${value}`);
+    }
+
+    private _save() {
+        Gio.Settings.sync(); // Necessary: http://stackoverflow.com/questions/9985140
+    }
+
+    private _getSchema(schemaId?: string) {
+        // https://gitlab.gnome.org/GNOME/gnome-shell/-/blob/gnome-43/js/misc/extensionUtils.js#L211
+        if (!schemaId)
+            schemaId = Self.metadata['settings-schema'];
+
+
+        // Expect USER extensions to have a schemas/ subfolder, otherwise assume a
+        // SYSTEM extension that has been installed in the same prefix as the shell
+        const schemaDir = Self.dir.get_child('schemas');
+        let schemaSource;
+        const schemaPath = schemaDir.get_path();
+        if (schemaDir.query_exists(null) && schemaPath !== null) {
+            schemaSource = Gio.SettingsSchemaSource.new_from_directory(schemaPath,
+                Gio.SettingsSchemaSource.get_default(),
+                false);
+        } else {
+            schemaSource = Gio.SettingsSchemaSource.get_default();
+        }
```

Comment:
Do we? This is a copy of the internal schema finding mechanism, so assuming by the comments I expect system-wide installations having their schema located at a different path.

Reply:
We package the schemas folder with the app. I thought that any standard installation would have the packaged schema and we can expect this to folder to exist. 

But we can leave the code as is to be safe. Also, I don't know how this will be handled in the future when we eventually do not ship compiled schemas anymore.

Reply:
I had a look at the AUR package for gsconnect:
* The schema will be in `/usr/share/glib-2.0/schemas/….gschema.xml`
* The extension itself will be in `/usr/share/gnome-shell/extensions/…`

Reply:
Ah, sorry, now I understand your point!